pkg/workload/workload.go

```diff
@@ -1062,14 +1080,53 @@ func AdmissionChecksForWorkload(log logr.Logger, wl *kueue.Workload, admissionCh
 	return acNames
 }
 
-func Evict(ctx context.Context, c client.Client, recorder record.EventRecorder, wl *kueue.Workload, reason, underlyingCause, msg string, clock clock.Clock) error {
-	evictionReason := reason
-	if reason == kueue.WorkloadDeactivated && underlyingCause != "" {
-		evictionReason = fmt.Sprintf("%sDueTo%s", evictionReason, underlyingCause)
+type EvictionOption func(*EvictionOptions)
+
+type EvictionOptions struct {
+	CustomPrepare func() (*kueue.Workload, error)
+}
+
+func DefaultEvictionOptions() *EvictionOptions {
+	return &EvictionOptions{
+		CustomPrepare: nil,
 	}
-	prepareForEviction(wl, clock.Now(), evictionReason, msg)
-	reportWorkloadEvictedOnce := workloadEvictionStateInc(wl, reason, underlyingCause)
-	if err := ApplyAdmissionStatus(ctx, c, wl, true, clock); err != nil {
+}
+
+func WithCustomPrepare(customPrepare func() (*kueue.Workload, error)) EvictionOption {
+	return func(o *EvictionOptions) {
+		if customPrepare != nil {
+			o.CustomPrepare = customPrepare
+		}
+	}
+}
+
+func Evict(ctx context.Context, c client.Client, recorder record.EventRecorder, wlOrig *kueue.Workload, reason, underlyingCause, msg string, clock clock.Clock, options ...EvictionOption) error {
```

Comment:
The `underlyingCause` argument also not required. I'm wondering if we can move it to EvictOptions as well. It can be implemented in follow-up PR.